### PR TITLE
[webview_flutter] Add setCookie to CookieManager and support initial cookies in creation params.

### PR DESCRIPTION
**Pull request by BeMacized**
This PR adds a function for setting cookies to the CookieManager, as well support for setting initial cookies through the `CreationParams` object.

~~As this PR depends on, and contains the changes of, updates made to the platform interface in #4555, as well as the platform specific implementations in #4556 and #4557, this PR will remain in draft mode until those changes have been merged and published.~~

Fixes flutter/flutter#27597

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter/lib/src/webview.dart [hunk 1]
```diff
@@ -721,16 +729,34 @@ class CookieManager {
...
+      }
+    }
+  }
 
   static CookieManager? _instance;
 
   /// Clears all cookies for all [WebView] instances.
   ///
-  /// This is a no op on iOS version smaller than 9.
+  /// This is a no op on iOS versions below 9.
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter/lib/src/webview.dart:749]: We don't support versions below 9 any more, so you can just remove this.

**Review by stuartmorgan — COMMENTED** [review]
Drive-by comment nit.

**Comment by stuartmorgan** [comment]
Dependencies are published; this is ready to be finalized.

File: packages/webview_flutter/webview_flutter/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,8 @@
+## 2.7.0
+
+* Added `setCookie` to CookieManager.
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter/CHANGELOG.md:3]: Nit: Adds

File: packages/webview_flutter/webview_flutter/pubspec.yaml [hunk 3]
```diff
@@ -19,9 +19,11 @@ flutter:
 dependencies:
   flutter:
     sdk: flutter
-  webview_flutter_android: ^2.7.0
+  webview_flutter_android: ^2.8.0
   webview_flutter_platform_interface: ^1.8.0
-  webview_flutter_wkwebview: ^2.5.0
+# TODO(BeMacized): Update dependency once published
+  webview_flutter_wkwebview:
+    path: ../webview_flutter_wkwebview
```
Comment (stuartmorgan) [inline packages/webview_flutter/webview_flutter/pubspec.yaml:26]: This didn't get resolved; should be 2.6.0

**Review by stuartmorgan — CHANGES_REQUESTED** [review]
Just minor things; looks good overall.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by stuartmorgan** [comment]
`submit-queue` is incorrectly reporting red; overriding.

**Comment by ismanong** [comment]
Cookies are encoded by default using UrlEncode, which is inconsistent with browser behavior and can cause some possible business problems.

**Comment by stuartmorgan** [comment]
This PR was landed last year; it can't be changed. If you have an issue to report, please use the issue tracker.

**Pull request merged**